src/libs/API.js

```diff
@@ -1,237 +1,24 @@
-        }
-        Log.info('Finished API request', false, logParams);
-    }
-
-    if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
-        const credentials = NetworkStore.getCredentials();
-
-        // Credentials haven't been initialized. We will not be able to re-authenticate with the API
-        const unableToReauthenticate = (!credentials || !credentials.autoGeneratedLogin
-            || !credentials.autoGeneratedPassword);
-
-        // There are some API requests that should not be retried when there is an auth failure like
-        // creating and deleting logins. In those cases, they should handle the original response instead
-        // of the new response created by handleExpiredAuthToken.
-        const shouldRetry = lodashGet(queuedRequest, 'data.shouldRetry');
-        if (!shouldRetry || unableToReauthenticate) {
-            // Check to see if queuedRequest has a resolve method as this could be a persisted request which had it's promise handling logic stripped
-            // from it when persisted to storage
-            if (queuedRequest.resolve) {
-                queuedRequest.resolve(response);
-            }
-            return;
-        }
-
-        handleExpiredAuthToken(queuedRequest.command, queuedRequest.data, queuedRequest.type)
-            .then(queuedRequest.resolve || (() => Promise.resolve()))
-            .catch(queuedRequest.reject || (() => Promise.resolve()));
-        return;
-    }
-
-    // Check to see if queuedRequest has a resolve method as this could be a persisted request which had it's promise handling logic stripped
-    if (!queuedRequest.resolve) {
-        return;
-    }
+import * as Middleware from './Middleware';
+import CONST from '../CONST';
 
-    // All other jsonCode besides 407 are treated as a successful response and must be handled in the .then() of the API method
-    queuedRequest.resolve(response);
-});
+// Setup API middlewares. Each request made will pass through a series of middleware functions that will get called in sequence (each one passing the result of the previous to the next).
+// Logging - Logs request details and errors.
+Request.use(Middleware.Logging);
 
-/**
- * @param {Object} parameters
- * @param {Boolean} [parameters.useExpensifyLogin]
- * @param {String} parameters.partnerName
- * @param {String} parameters.partnerPassword
- * @param {String} parameters.partnerUserID
- * @param {String} parameters.partnerUserSecret
- * @param {String} [parameters.twoFactorAuthCode]
- * @param {String} [parameters.email]
- * @param {String} [parameters.authToken]
- * @returns {Promise}
- */
-function Authenticate(parameters) {
-    const commandName = 'Authenticate';
+// Recheck - Sets a timeout timer for a request that will "recheck" if we are connected to the internet if time runs out. Also triggers the connection recheck when we encounter any error.
+Request.use(Middleware.Recheck);
```

Comment:
Suggestion: rename to `Middleware.RecheckConnection`

This just gives it a little more context. When I first saw this, I wasn't sure what "recheck" meant at all.

Reply:
yep, great suggestion